### ISSUE TITLE
Unexpected Error after rebooting for driver change

### ISSUE THREAD

**Issue by Samutz** [sirWest/AudioSwitch#18]
While troubleshooting some graphics card issues, I had to disable the graphics drivers so that Windows would use the default. After the reboot, Audioswitch doesn't start. After re-enabling the drivers, it still doesn't start. Here is the log:

```
Log Entry: 7/21/2015 8:12:16 PM

System.InvalidOperationException: There is an error in XML document (0, 0). ---> System.Xml.XmlException: Root element is missing.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlTextReader.Read()
   at System.Xml.XmlReader.MoveToContent()
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderSettings.Read12_Settings()
   --- End of inner exception stack trace ---
   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, XmlDeserializationEvents events)
   at System.Xml.Serialization.XmlSerializer.Deserialize(TextReader textReader)
   at AudioSwitch.Classes.Settings.Load()
   at AudioSwitch.Classes.Program.Main(String[] args)
-------------------------------
```

It appears that Settings.xml is being overwritten with about 800 null bytes. Deleting the Settings.xml file causes another error:

```
Log Entry: 7/21/2015 8:16:40 PM

System.IO.FileNotFoundException: Could not find file 'C:\Users\Samutz\AppData\Local\AudioSwitch\Settings.xml'.
File name: 'C:\Users\Samutz\AppData\Local\AudioSwitch\Settings.xml'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path)
   at AudioSwitch.Classes.Settings.Load()
   at AudioSwitch.Classes.Program.Main(String[] args)
-------------------------------
```

I have to reinstall the program to create a new Settings.xml, and of course lose my previous settings.
This all happened once before when I initially installed my video card too. I suspect something is going wrong when the audio drivers for the HDMI ports are being changed.

I don't know how to fix it writing null bytes to the Settings.xml file, but to alleviate the problem I suggest adding catches for these exceptions and writing a new Settings.xml when they are caught.

**Comment by sirWest** [comment]
indeed, all my previous efforts fixing this issue are not obviously working :( I'll put default settings to AudioSwitch and make it create a new settings file, there really doesn't seem any other solution to this. XML Serializer class seems to run into some issues that can only be in .NET itself - I have that kind of code in all my work related projects and it never fails like this, so I just can't pin the problem it has with AudioSwitch. I will fix this in next 2.1.1 bugfix release.

**Comment by sirWest** [comment]
the configuration file got corrupted, this error has been fixed several versions ago
